distribution/helm/vector-agent/templates/daemonset.yaml

```diff
@@ -14,9 +14,7 @@ spec:
   template:
     metadata:
       annotations:
-        {{- if not .Values.externalConfigMap }}
-        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```

Comment:
@MOZGIII shouldn't this annotation cause the pods to require a redeploy on any change in Values that causes the configmap to change?

Reply:
No, this only sum changes when the *template* changes, unfortunately. It's reading the raw template file and just sums it.

Reply:
Hmm, maybe a better fix would be to evaluate that template instead?

Reply:
Oh, word - didn't realize that!

Reply:
I think either hashing _just_ the values that affect the configuration or hashing the final configuration would probably be optimal

Reply:
Yep, fixed it, and disabled the `podValuesChecksumAnnotation` by default.

The code we had there is from the Helm chart tips and tricks docs. Well, I guess they were just confused with their own tool, happens.

Reply:
Is `podValuesChecksum` still needed if the configmap based checksum properly works?